text/0000-no-panic-abs.md

```diff
@@ -0,0 +1,86 @@
+- Feature Name: no_panic_abs
+- Start Date: 2016-07-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add the non-panicking `checked_abs`, `overflowing_abs` and `wrapping_abs` 
+functions to all signed integer types.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, calling `abs()` on one of the signed integer types might panic (in 
+debug mode at least) because the absolute value of the largest negative value 
+can not be represented in that signed type. Unlike all other integer 
+operations, there is currently not a non-panicking version on this function.
+
+# Detailed design
+[design]: #detailed-design
+
+This is the current implementation of `abs()`:
+
+```rust
+pub fn abs(self) -> Self {
+	if self.is_negative() {
+		-self
+	} else {
+		self
+	}
+}
+```
+
+This RFC proposes to add the following:
+
+```rust
+pub fn checking_abs(self) -> Option<Self> {
```

Comment:
`checked_abs`, not `checking`, would seem to be more consistent with existing conventions.